src/hooks/safe-apps/useRankedSafeApps.ts

```diff
@@ -1,30 +1,27 @@
-import { useMemo } from 'react'
-import { sampleSize } from 'lodash'
+import { difference, sampleSize } from 'lodash'
 import type { SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
 import { getAppsUsageData, rankSafeApps } from '@/services/safe-apps/track-app-usage-count'
+import { FEATURED_APPS_TAG } from '@/components/dashboard/FeaturedApps/FeaturedApps'
 
-// number of ranked Safe Apps that we include in the array
+// number of ranked Safe Apps that we want to display
 const NUMBER_OF_SAFE_APPS = 5
 
 const useRankedSafeApps = (safeApps: SafeAppData[], pinnedSafeApps: SafeAppData[]): SafeAppData[] => {
-  return useMemo(() => {
-    if (!safeApps.length) return []
+  if (!safeApps.length) return []
 
-    const usageSafeAppsData = getAppsUsageData()
-    const mostUsedSafeAppsIds = rankSafeApps(usageSafeAppsData, pinnedSafeApps).slice(0, NUMBER_OF_SAFE_APPS)
-    const mostUsedSafeApps = mostUsedSafeAppsIds
-      .map((id) => safeApps.find((safeApp) => String(safeApp.id) === id))
-      .filter(Boolean) as SafeAppData[]
+  const nonFeaturedApps = safeApps.filter((app) => !app.tags?.includes(FEATURED_APPS_TAG))
 
-    // we add random Safe Apps if no enough Safe Apps are present
-    const numberOfRandomSafeApps = NUMBER_OF_SAFE_APPS - mostUsedSafeApps.length
-    const nonRankedApps = safeApps.filter((app) => !mostUsedSafeAppsIds.includes(String(app.id)))
-    const randomSafeApps = sampleSize(nonRankedApps, numberOfRandomSafeApps)
+  const usageSafeAppsData = getAppsUsageData()
+  const mostUsedSafeAppsIds = rankSafeApps(usageSafeAppsData)
+  const mostUsedSafeApps = mostUsedSafeAppsIds
+    .map((id) => nonFeaturedApps.find((safeApp) => String(safeApp.id) === id))
+    .filter(Boolean) as SafeAppData[]
```

Comment:
I would replace this with 
```
.filter((app) => !app.tags?.includes(FEATURED_APPS_TAG))
```

Reply:
I tried removing the filter but it seems like `difference` is not filtering out `undefined` items. What did you have in mind?

Reply:
I meant the filter by featured.

Reply:
I moved the filter call to the end of the function as discussed but noticed that it could lead to some edge cases e.g. if there are no pinned and ranked safe apps and we randomly select 5 safe apps and by chance one of them is a featured app it would lead to fewer than 5 random apps being displayed.

Reply:
Well, we should select 7 random apps then, not 5.